minisat/core/SolverTypes.h

```diff
@@ -219,17 +230,42 @@ class Clause {
     const Lit&   last        ()      const   { return data[header.size-1].lit; }
 
     bool         reloced     ()      const   { return header.reloced; }
+#if SIZE_MAX == UINT32_MAX
     CRef         relocation  ()      const   { return data[0].rel; }
     void         relocate    (CRef c)        { header.reloced = 1; data[0].rel = c; }
+#else
+#if defined __clang__
+  #pragma clang diagnostic push
+#elif defined __GNUC__
+  #pragma GCC diagnostic push
+#elif defined _MSC_VER
+  #pragma warning(push)
+#endif
+#if defined __clang__
+  #pragma clang diagnostic ignored "-Wstrict-aliasing"
```

Comment:
Could you please hide these defines in a single one, e.g. IGNORE_ALIASING_WARNING (and eventually also modify the other occurrences)? Having a new file for this, e.g. mtl/Defines.h, should be a good start.